samples/boards/stm32/usbc/sink/CMakeLists.txt

```diff
@@ -0,0 +1,9 @@
+# SPDX-License-Identifier: Apache-2.0
```

Comment:
Why is it `samples/boards/stm32/usbc ` and not `samples/subsys/usbc`? Does it not use generic USBC API?

Reply:
I initially had it in  samples/subsys/usbc but it was recommended that I move it to samples/boards/stm32/usbc because dev-board specific files were needed to configure the board for USBC operation.

Reply:
It would be better to have this sample in `samples/subsys/usbc`. One might think the USBC driver API is not generic enough. Besides, in the sample itself there is not much board specific. [stm32g081b_eval_board.c](https://github.com/zephyrproject-rtos/zephyr/pull/41856/commits/2ab55ab8928652510ce867d8f357925523b40a12#diff-600bd7c0d7922f99cba8131489cbdc91723bda56fbc258c1e6db7eb6c8e8e8c5) uses generic ADC driver API as I see and could be abstracted further.

Reply:
I initially had it there but it was recommended that I move it to boards. I'm inclined to leave this sample here because I have another PR that implements the full USBC subsystem stack that included a more general sample.

Reply:
> but it was recommended that I move it to boards.

No, it was recommended that this sample was not board centric in its code, but instead be made generic.
https://github.com/zephyrproject-rtos/zephyr/pull/41856#discussion_r796752627
> Generally we should avoid such constructs, especially because it makes it harder to reuse a sample on another board which shares same SoC or peripheral as those boards.
>
> Is it not possible to select source file based on devicetree knowledge or Kconfig setting ?
>
> And if not, what makes this sample so special it should be restricted to those board, and not a general restriction based on hw capabilities.

and then later:
https://github.com/zephyrproject-rtos/zephyr/pull/41856#discussion_r797578264
> Sure, but that's still not a reason to be board centric.


Only as an alternative solution to move it:
> If this really is a board specific sample (which I don't think from the look of it),

So i'm aligned with @jfischer-no on his view to be optimal solution, although I do accept the `samples/boards` approach if code is board centric.

Reply:
> I initially had it there but it was recommended that I move it to boards. I'm inclined to leave this sample here because I have another PR that implements the full USBC subsystem stack that included a more general sample.

I also assumed that it would be followed by an USBC stack. Nevertheless, you may be putting stones in your own way. A sample in `samples/subsys` can be revised/simplified when USBC stack is ready, sample in `samples/boards/stm32` must probably be removed.